src/Compiler/CompilerDialects.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------------------ CompilerDialects.cpp ------------------------===//
+
+#include "CompilerDialects.hpp"
+
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+#include "src/Dialect/ONNX/ONNXDialect.hpp"
+
+#include "mlir/InitAllDialects.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+DialectRegistry registerDialects(ArrayRef<accel::Accelerator::Kind> accels) {
+  DialectRegistry registry;
+
+  // Note that we cannot consult command line options because they have not yet
+  // been parsed when registerDialects() is called.
+
+  registry.insert<arith::ArithDialect>();
+  registry.insert<linalg::LinalgDialect>();
+  registry.insert<affine::AffineDialect>();
+  registry.insert<LLVM::LLVMDialect>();
+  registry.insert<scf::SCFDialect>();
+  registry.insert<func::FuncDialect>();
+  registry.insert<vector::VectorDialect>();
+  registry.insert<shape::ShapeDialect>();
+  registry.insert<math::MathDialect>();
+  registry.insert<memref::MemRefDialect>();
+  registry.insert<ONNXDialect>();
+  registry.insert<KrnlDialect>();
```

Comment:
In the past, we have tried to make it possible for folks to build onnx-mlir without needing Krnl, as it is somewhat specialized for our use. Would it make sense to keep this separation?

Reply:
this PR doesn't change this, right?

I don't think the restructuring in this PR would make it harder to implement the separation if we want to do that in the future

Reply:
> build onnx-mlir without needing Krnl

if you want to use onnx-mlir as a library and use the ONNX dialect without Krnl, then I believe that the recommended way (at least what Groq does) is to write your own version of the onnx-mlir/onnx-mlir-opt executables and also your own registerDialects(), registerPasses(), addPasses(), etc; that is, just link things like OMONNXOps (dialect, canonicalization) and OMONNXRewrite (shape inference, const prop, decomposition) and build up things from there

in other words, registerDialects() is only meant to be reusable between onnx-mlir and onnx-mlir-opt